### PR TITLE
Docs for List

### PR DESCRIPTION
**Pull request by aspeddro**
- Add examples header `## Examples`
- Format code blocks
- Remove indentation
- I rewrote some comments

### CODE REVIEW

File: src/Core__List.resi [hunk 1]
```diff
@@ -23,129 +23,158 @@
...
+
+```rescript
+List.length(list{1, 2, 3}) // 3
+```
 */
 let length: t<'a> => int
 
-/** **See** [`length`](##length) */
+/**
+`size(list)`. **See** [`length`](##length)
```
Comment (aspeddro) [inline src/Core__List.resi:49]: There are several comments using this format:

```
[`length`](##length)
```

In VSCode this is a link that opens a file that doesn't exist.

image


How should we reference functions?

Comment (aspeddro) [reply src/Core__List.resi:49]: A proposal:

```
[`function`](ModuleName.function)
```

For function in same module:

```
[`function`](function)
```

Comment (zth) [reply src/Core__List.resi:49]: Right, that's an internal markdown link. We haven't 100% landed on the format yet (it's dependent on the documentation extraction, which we're working on in parallel). Let's revisit this a bit later, it's easy to fix once we have the format we want down.

File: src/Core__List.resi [hunk 2]
```diff
@@ -23,129 +23,158 @@
...
+List.head(list{}) // None
+List.head(list{1, 2, 3}) // Some(1)
+```
 */
 let head: t<'a> => option<'a>
 
 /**
-  Same as [head](#head), but raises an exception if `someList` is empty. Use
-  with care.
+`headExn(list)` same as [`head`](#head), but raises an exception if `list` is
```
Comment (zth) [inline src/Core__List.resi:73]: Should we add the same `## Exceptions` section here that you added in another PR?

**Review by zth — APPROVED** [review]
Just a comment!

File: src/Core__List.resi (hunk 2 above)
Comment (aspeddro) [reply src/Core__List.resi]: Done

File: src/Core__List.resi (hunk 1 above)
Comment (aspeddro) [reply src/Core__List.resi]: Ok

**Review by zth — APPROVED** [review]

**Pull request merged**